### PR TITLE
feat(expr): add local make_timestamptz()

### PR DESCRIPTION
**Pull request by KeXiangWang**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Support make_timestamptz() partially. Currently only local mode is supported, distributed and streaming modes are not supported yet.
Corresponding docs in PgSQL: https://www.postgresql.org/docs/current/functions-datetime.html

## Checklist

- [ ] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

### CODE REVIEW

File: e2e_test/batch/types/timestamptz_utc.slt.part [hunk 1]
```diff
@@ -165,3 +168,67 @@ query I
 select count('2022-10-01 12:00:00-08:00'::timestamp with time zone);
 ----
 1
+
+query T
+SELECT make_timestamptz(1973, 07, 15, 08, 15, 55.33);
```
Comment (xiangjinwu) [inline e2e_test/batch/types/timestamptz_utc.slt.part:173]: We can put these in the `local` folder before they are supported in distributed mode.

https://github.com/risingwavelabs/risingwave/blob/483347725eacbe39a9a60b0817ac145aaf205f80/e2e_test/batch/local_mode.slt#L8

File: src/frontend/src/expr/function_impl/make_timestamptz.rs [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+) -> Result<Timestamptz> {
+    let time_zone = Timestamptz::lookup_time_zone(time_zone).map_err(time_zone_err)?;
+    if !sec.is_finite() {
+        return Err(ExprError::InvalidParam {
+            name: "sec",
+            reason: "invalid sec".into(),
+        });
+    }
+    let sec_u32 = sec.0.trunc() as u32;
+    let microsecond_u32 = ((sec.0 - sec.0.trunc()) * 1_000_000.0).round() as u32;
```
Comment (xiangjinwu) [inline src/frontend/src/expr/function_impl/make_timestamptz.rs:75]: `round` -> `round_ties_even`, or just use `From<F64> for i32`

What if any of the input is negative?

File: src/frontend/src/expr/function_impl/make_timestamptz.rs [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        NaiveTime::from_hms_micro_opt(hour as u32, min as u32, sec_u32, microsecond_u32)
+            .ok_or_else(|| ExprError::InvalidParam {
+                name: "hour, min, sec",
+                reason: "invalid time".into(),
+            })?,
+    );
+    let date_time = naive_date_time
+        .and_local_timezone(time_zone)
+        .latest()
+        .ok_or_else(|| ExprError::InvalidParam {
```
Comment (xiangjinwu) [inline src/frontend/src/expr/function_impl/make_timestamptz.rs:92]: When this function is moved from frontend to expr_impl after batch/stream support, we can reuse `timestamp_at_time_zone`. Right?

Comment (KeXiangWang) [reply src/frontend/src/expr/function_impl/make_timestamptz.rs:92]: Yes

File: src/frontend/src/expr/function_impl/make_timestamptz.rs (hunk 2 above)
Comment (KeXiangWang) [reply src/frontend/src/expr/function_impl/make_timestamptz.rs:75]: Updated with `round_ties_even `. 

For `year`, negative number is acceptable.
For other fields, `NaiveDate::from_ymd_opt` and `NaiveTime::from_hms_micro_opt` handle the negative numbers and return err.
`sec.0.trunc() as u32;` will convert negative float to 0, causing unexpected output. So now, if negative sec is found, error is returned directly.
With the new commit, more tests are included to cover negative nums.

File: e2e_test/batch/types/timestamptz_utc.slt.part (hunk 1 above)
Comment (KeXiangWang) [reply e2e_test/batch/types/timestamptz_utc.slt.part]: Thanks! updated

**Review by xiangjinwu — APPROVED** [review]

**Pull request merged**